app/src/main/java/com/github/sdpcoachme/CoachesListActivity.kt

```diff
@@ -28,23 +28,30 @@ import androidx.compose.ui.text.style.TextOverflow
 import androidx.compose.ui.unit.dp
 import com.github.sdpcoachme.data.UserInfo
 import com.github.sdpcoachme.ui.theme.CoachMeTheme
+import kotlinx.coroutines.future.await
 import java.util.concurrent.CompletableFuture
 
 class CoachesListActivity : ComponentActivity() {
+    var stateLoading = CompletableFuture<Void>()
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+        // TODO: update this to be current device location
+        val currentLat = 46.519054480712015
+        val currentLong = 6.566757578464391
         val futureListOfCoaches = (application as CoachMeApplication).database
-            .getAllUsers().thenApply {
+            .getAllUsersByNearest(
+                latitude = currentLat,
+                longitude = currentLong
+            ).thenApply {
                 it.filter { user -> user.coach }
             }
+
         setContent {
             var listOfCoaches by remember { mutableStateOf(listOf<UserInfo>()) }
 
-            // TODO: Need to handle the future correctly in cases like this (might need to use coroutines)
-            var f by remember { mutableStateOf(futureListOfCoaches) }
-            f.thenAccept {
-                listOfCoaches = it
-                f = CompletableFuture.completedFuture(null)
+            LaunchedEffect(true) {
+                listOfCoaches = futureListOfCoaches.await()
+                stateLoading.complete(null)
             }
```

Comment:
I would put a small comment before LaunchedEffect to explain the purpose for testing.